docs/content/developer/tutorials/sponsored-txs.mdx

```diff
@@ -0,0 +1,738 @@
+iota-keys = { git = "https://github.com/iotaledger/iota", package = "iota-keys" }
+shared_crypto = { git = "https://github.com/iotaledger/iota", package = "shared-crypto" }
+
+tokio = { version = "1.2", features = ["full"] }
+anyhow = "1.0"
+bcs = "0.1.4"
+clap = "3.0.0-beta.2"
+```
+
+and in the `src/main.rs` file, add the logic for `construct_tx` function:
+
+```rust
+async fn construct_tx(
+    keystore_path: &str,
+    encoded_tx: &str,
+    encoded_sponsor_sig: &str,
+    sender_addr: &str,
+) -> Result<(), anyhow::Error> {
+    let keystore = FileBasedKeystore::new(&PathBuf::from(keystore_path))
+        .map_err(|e| anyhow::anyhow!("Failed to load keystore: {:?}", e))?;
+
+    let decoded_tx = fastcrypto::encoding::Base64::decode(encoded_tx)
+        .map_err(|_| anyhow::anyhow!("Failed to decode transaction"))?;
+    let decoded_sig = fastcrypto::encoding::Base64::decode(encoded_sponsor_sig)
+        .map_err(|_| anyhow::anyhow!("Failed to decode sponsor's signature"))?;
+
+    let tx: TransactionData = bcs::from_bytes(&decoded_tx)
+        .map_err(|_| anyhow::anyhow!("Failed to deserialize transaction data"))?;
+    let sig_bcs = Signature::from_bytes(&decoded_sig)
+        .map_err(|_| anyhow::anyhow!("Failed to deserialize sponsor's signature"))?;
+
+    let intent_msg = IntentMessage::new(Intent::iota_transaction(), tx.clone());
+
+    let sender_addr = IotaAddress::from_str(sender_addr)
+        .map_err(|_| anyhow::anyhow!("Failed to parse sender address"))?;
+
+    let sender_sig = keystore
+        .sign_secure(&sender_addr, &tx, intent_msg.intent)
+        .map_err(|_| anyhow::anyhow!("Failed to sign transaction with sender's keystore"))?;
+
+    let signed_tx = iota_sdk::types::transaction::Transaction::from_generic_sig_data(
+        intent_msg.value,
+        vec![
+            iota_sdk::types::signature::GenericSignature::Signature(sender_sig),
+            iota_sdk::types::signature::GenericSignature::Signature(sig_bcs),
+        ],
+    );
+
+    let iota_testnet = IotaClientBuilder::default()
+        .build_testnet()
+        .await
+        .map_err(|_| anyhow::anyhow!("Failed to connect to IOTA testnet"))?;
+
+    let transaction_block_response = iota_testnet
+        .quorum_driver_api()
+        .execute_transaction_block(
+            signed_tx.clone(),
+            iota_sdk::rpc_types::IotaTransactionBlockResponseOptions::full_content()
+                .with_raw_input(),
+            iota_sdk::types::quorum_driver_types::ExecuteTransactionRequestType::WaitForLocalExecution,
+        )
+        .await
+        .map_err(|e| anyhow!("Failed to execute transaction: {:?}", e))?;
+
+    println!("{:?}", transaction_block_response);
+    Ok(())
+}
+```
+
+`construct_tx` function will try to decode the provided transaction bytes and sponsor's signature. Then, it will sign the transaction with the sender's keystore and send the transaction to the IOTA network and print the response.
```

Comment:
```suggestion
The `construct_tx` function will try to decode the provided transaction bytes and the sponsor's signature. Then, it will sign the transaction with the sender's keystore, send it to the IOTA network, and print the response.
```